### ISSUE TITLE
[E2E] test-e2e should allow running test against testbed where only one datastore is present.

### ISSUE THREAD

**Issue by shuklanirdesh82** [vmware/docker-volume-vsphere#1463]
Test execution should fail gracefully instead of `invalid memory address or nil pointer dereference`

```
=> Running target test-e2e-runalways Thu Jun 22 17:41:22 UTC 2017

GO15VENDOREXPERIMENT=1 go test -v -timeout 30m -tags runalways github.com/vmware/docker-volume-vsphere/tests/e2e 
=== RUN   Test

----------------------------------------------------------------------
PANIC: volumeproperty_test.go:78: VolumePropertyTestSuite.TearDownSuite

... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0x4582EC)

/usr/local/go/src/runtime/panic.go:458
  in gopanic
/usr/local/go/src/runtime/panic.go:62
  in panicmem
/usr/local/go/src/runtime/sigpanic_unix.go:24
  in sigpanic
volumeproperty_test.go:79
  in VolumePropertyTestSuite.TearDownSuite
/usr/local/go/src/reflect/value.go:302
  in Value.Call
/usr/local/go/src/runtime/asm_amd64.s:2086
  in goexit
--- FAIL: Test (5.72s)
FAIL
OOPS: 0 passed, 1 skipped, 1 FIXTURE-PANICKED
exit status 1
FAIL	github.com/vmware/docker-volume-vsphere/tests/e2e	5.728s
make: *** [test-e2e-runalways] Error 1
make: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'
make[1]: *** [test-e2e] Error 2
make: *** [test-e2e] Error 2
```

**Comment by shuklanirdesh82** [comment]
minor improvement is required at inputparams.go
1. need to fix condition `len(config.Datastores) < 1 {` (test-e2e should run against 1datastsore setup and let the test to decide whether stop execution incase of insufficient datastore.

2. log the error message about insufficient datastore

```
if config.DockerHostNames[0] == "" || config.DockerHostNames[1] == "" ||
		len(config.Datastores) < 1 {
		return nil
	}
```

**Comment by ashahi1** [comment]
Fixed as part of #1530
